### DISCUSSION TITLE
Step fails with exit code 4 even though GH_TOKEN is set

### DISCUSSION THREAD

**Discussion by junefish** [community/community#128276] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm getting this error when I try to create a PR within my Action:
```
gh: To use GitHub CLI in a GitHub Actions workflow, set the GH_TOKEN environment variable. Example:
  env:
    GH_TOKEN: ${{ github.token }}
Error: Process completed with exit code 4.
```

The [documentation](https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows) gives a different format for the expected env declaration:
```
env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
I've tried both formats in my YAML file, and both give me the same error.

Here's the full step from YAML file:
```
- name: Create PR for any article ID updates
        run: |
          git pull
          TEMP=$(mktemp -u XXXX)
          BRANCH="add-article-ids-$TEMP"
          git switch -c $BRANCH -m
          git add troubleshooting/*.mdx
          git status -s
          git commit --untracked-files=no -m "[automated] update new articles with articleIDs"
          git push --set-upstream origin $BRANCH
          gh pr create -d -a "junefish" -t "[automated] update article IDs" \
            -b "This pull request has been automatically generated by a Github Action. Please review that all changes are correct."
          env:
            GH_TOKEN: ${{ github.token }}
```

I'm really stumped on why this is failing since I am setting `GH_TOKEN`.

**Comment by vit0rrr** [comment]
I'm Brazilian, so please excuse the grammar mistakes

1. Check Your YAML Structure
Make sure that the GH_TOKEN variable is being defined correctly and in the correct place in your YAML file. The GH_TOKEN environment variable should be outside the run block.

2. Fix Your YAML Syntax
Here is a corrected and more detailed version of your YAML file:

yaml
Copy Code
name: Create PR for Article ID Updates

on:
push:
branches:
- main

jobs:
create-pr:
runs-on: ubuntu-latest
steps:
- name: Checkout repository
uses: actions/checkout@v2

- name: Set up Git
run: | git config --global user.name 'GitHub Actions'
git config --global user.email 'actions@github.com'

- name: Pull latest changes
run: git pull

- name: Create new branch and commit changes
run: | TEMP=$(mktemp -u XXXX)
BRANCH="add-article-ids-$TEMP"
git switch -c $BRANCH -m
git add troubleshooting/*.mdx
git status -s
git commit --untracked-files=no -m "[automated] update new articles with articleIDs"
git push --set-upstream origin $BRANCH

- name: Create PR
env:
GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
run: | gh pr create -d -a "junefish" -t "[automated] update article IDs" \
-b "This pull request was automatically generated by a Github Action. Please verify that all changes are correct."
3. Ensure GitHub CLI is Installed
Make sure GitHub CLI (gh) is installed and available in the runtime. You can add a step to install GitHub CLI before using it:

yaml
Copy code
- name: Install GitHub CLI
run: sudo apt-get install gh

4. Ensure Token is Available
The GITHUB_TOKEN token is automatically provided by GitHub Actions, but it is a good practice to ensure that it is correctly available. In the YAML file above, use secrets.GITHUB_TOKEN to set the GH_TOKEN.

5. Other Considerations
Token Permissions: Make sure the token has the necessary permissions to create pull requests. In GitHub Actions, the default token (GITHUB_TOKEN) should have these permissions, but if you're using a custom token, check the permissions.
Specific error: If the issue persists, exit code 4 may be related to a GitHub CLI-specific failure. Check the documentation or run the command in more detail (gh pr create --help) for more information.

**Comment by spenserblack** [comment, marked as answer]
> Here's the full step from YAML file:
> 
> ```
> - name: Create PR for any article ID updates
>         run: |
>           git pull
>           TEMP=$(mktemp -u XXXX)
>           BRANCH="add-article-ids-$TEMP"
>           git switch -c $BRANCH -m
>           git add troubleshooting/*.mdx
>           git status -s
>           git commit --untracked-files=no -m "[automated] update new articles with articleIDs"
>           git push --set-upstream origin $BRANCH
>           gh pr create -d -a "junefish" -t "[automated] update article IDs" \
>             -b "This pull request has been automatically generated by a Github Action. Please review that all changes are correct."
>           env:
>             GH_TOKEN: ${{ github.token }}
> ```

Based on this it looks like `env` is indented too much, and is actually part of the string value for the `run:` key. Try un-indenting it a bit.

**Reply by junefish** [reply]
This was it! I feel very dumb but it's working now. Thank you!
